--- libifp-1.0.1.0/src/comp.c.orig
+++ libifp-1.0.1.0/src/comp.c
@@ -104,7 +104,11 @@
 
     i = ifp_list_dirs(dev, "\\", noop_dir_callbk, &n);
     if (i) {
-        ifp_err_i(i, "couldn't get basic directory listing.");
+	/* See Debian bug #331585 -- if this tests fails the user
+	   can't even format the device to fix it. Warn instead. -- piman */
+	n = 0;
+	ifp_wrn("couldn't get basic directory listing.");
+        /* ifp_err_i(i, "couldn't get basic directory listing."); */
         return i;
     }
 
